src/macros.rs

```diff
@@ -27,13 +27,18 @@ macro_rules! include_spv {
 /// The definition of this type is unstable.
 pub type Spirv = Align4<[u8]>;
 
-impl std::ops::Deref for Spirv {
-    type Target = [u32];
-    fn deref(&self) -> &[u32] {
-        #[allow(clippy::cast_ptr_alignment)]
-        unsafe {
-            std::slice::from_raw_parts(self.0.as_ptr() as *const u32, self.0.len() / 4)
+impl Spirv {
+    pub fn deref(&self) -> Vec<u32> {
+        let mut out = Vec::with_capacity(self.0.len() / 4);
```

Comment:
If we are going to allocate a vector anyway, there's always [`wgpu::read_spirv`](https://docs.rs/wgpu/0.4.0/wgpu/fn.read_spirv.html). And it will handle the endianness correctly, too.

Consequently, this is what was used before the macro: https://github.com/parasyte/pixels/blob/9f1afdd3b324ccbe68556bca36253449c2185cb7/src/lib.rs#L113-L120

We also don't need to use `Deref` if we are removing the `slice::from_raw_parts`.